### ISSUE TITLE
Intermittent 503 UH errors for external service calls 

### ISSUE THREAD

**Issue by keerthig9** [istio/istio#26369]
**Bug description**
We recently enabled Egress for external service calls in our cluster and all was working fine for a week, later we observed spikes of 503 UH errors for external service calls in envoy sidecar and recovered automatically. 2 days it was a small spike but one day the errors went up for nearly 4 hours causing availability drop for cluster and recovered automatically. We disabled egress as mitigation and now we are trying to root cause the issue and we didn't find way to reproduce.

Successful request log in istio-proxy:
```
[2020-07-27T18:10:57.205Z] "GET /products/xxx HTTP/1.1" 200 - "-" "-" 0 5754 11 11 "-" “xxx-1.5.0” "a76ec2c0-bc79-42df-a201-f1d29d26bc96" “externalservice.com:30445" “x.x.x.x:443" outbound|443|http--calls|externalservice.com - x.x.x.x:30445 x.x.x.x:59772 - -
```

Failed request log in istio-proxy:
```
[2020-07-27T18:10:57.252Z] "GET /products/xxx HTTP/1.1"  503 UH "-" "-" 0 19 0 - "-" “xxx-1.5.0” "95b75970-03a3-4548-9440-d9eff293de97" "externalservice.com:30445" "-" - - 10.0.31.98:30445 10.0.19.196:60416 - -
Note: In above logs for a request with 503 errors, outbound rule is not present.
```
We use ServiceEntry with VirtualService, DestinationRule to access  externalservice.com and no EgressGateway used.

**1) What causes Intermittent 503 UH errors for external service calls? **
**2) What are suggested steps to reproduce errors?**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**
NO 503 UH errors seen

**Steps to reproduce the bug**
NA

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
1.4.5

**How was Istio installed?**
helm

**Environment where bug was observed (cloud vendor, OS, etc)**
AWS eks

**Comment by keerthig9** [comment]
Seeing these logs in the proxy[Envoy (Epoch 0)] [2020-07-27 20:06:21.023][21][warning][config] [external/envoy/source/common/config/grpc_mux_subscription_impl.cc:82] gRPC config for type.googleapis.com/envoy.api.v2.Cluster rejected: Error adding/updating cluster(s) outbound|443|http-calls|externalservice.com: Invalid path: /etc/ssl-certs/cacerts.pem.

But everything is working before and after the event and event recovered automatically.

**Comment by howardjohn** [comment]
Is it feasible to try upgrading? https://istio.io/latest/news/support/announcing-1.4-eol-final/ This may have already been fixed, 1.4 is no longer supported

**Comment by keerthig9** [comment]
Before we upgrade, can you please list common cause for 503 UH errors, we need to debug the issue and cannot upgrade so easily. 

Along with that i have another question  "When we have two ServiceEntry for same host with different spec, which one will be picked by pilot? Will Istio merge the ServiceEntry config?"

**Comment by keerthig9** [comment]
> Seeing these logs in the proxy[Envoy (Epoch 0)] [2020-07-27 20:06:21.023][21][warning][config] [external/envoy/source/common/config/grpc_mux_subscription_impl.cc:82] gRPC config for type.googleapis.com/envoy.api.v2.Cluster rejected: Error adding/updating cluster(s) outbound|443|http-calls|externalservice.com: Invalid path: /etc/ssl-certs/cacerts.pem.
> 
> But everything is working before and after the event and event recovered automatically.

Found root cause for this, as we use volume mount for certificate above log in pilot is seen for a pod which was created much before certificate is installed in the cluster. So this is not the root cause and this error was just warning and didn't contribute to 503 errors and this is thrown for deployment pod which is not actively used. We need to continue investigation.

**Comment by keerthig9** [comment]
> Is it feasible to try upgrading? https://istio.io/latest/news/support/announcing-1.4-eol-final/ This may have already been fixed, 1.4 is no longer supported

Upgrading Istio to latest version takes time few months for us bcoz of operations tasks inside the project, we want to understand root cause, so that we can have sufficient tests to ensure we don't face this issue with 1.4.5 (Which we are using now) and even with Istio latest version when we upgrade. As 503 UH errors caused large availability issue for us in production environment, we want to root cause, reproduce and fix it if possible.

**Comment by howardjohn** [comment]
This looks like a duplicate of https://github.com/istio/istio/issues/26282
